cpp/src/parquet/schema.h

```diff
@@ -433,6 +433,8 @@ class PARQUET_EXPORT SchemaDescriptor {
   int ColumnIndex(const schema::Node& node) const;
 
   bool Equals(const SchemaDescriptor& other) const;
+  bool Equals(const SchemaDescriptor& other,
+              std::shared_ptr<std::stringstream> diff_msg) const;
```

Comment:
Three things: 1) this should take a raw pointer (no need to give ownership), 2) this should take a generic `ostream` not a specific implementation, 3) no need for two separate signatures as the second argument can be made optional
```suggestion
  bool Equals(const SchemaDescriptor& other,
              std::ostream* diff_output = NULLPTR) const;
```

Reply:
Thank you. I hope reviewing such noobie C++ coding isn't too taxing; I appreciate the help. :pray:

Reply:
It's definitely ok :-)